conans/test/unittests/client/build/cpp_std_flags_test.py

```diff
@@ -1,195 +1,261 @@
-        self.assertEqual(cppstd_flag("clang", "5", "gnu14"), '-std=gnu++14')
-        self.assertEqual(cppstd_flag("clang", "5", "17"), '-std=c++17')
-
-        self.assertEqual(cppstd_flag("clang", "5.1", "11"), '-std=c++11')
-        self.assertEqual(cppstd_flag("clang", "5.1", "14"), '-std=c++14')
-        self.assertEqual(cppstd_flag("clang", "5.1", "17"), '-std=c++17')
-
-        self.assertEqual(cppstd_flag("clang", "6", "11"), '-std=c++11')
-        self.assertEqual(cppstd_flag("clang", "6", "14"), '-std=c++14')
-        self.assertEqual(cppstd_flag("clang", "6", "17"), '-std=c++17')
-        self.assertEqual(cppstd_flag("clang", "6", "20"), '-std=c++2a')
-
-        self.assertEqual(cppstd_flag("clang", "7", "11"), '-std=c++11')
-        self.assertEqual(cppstd_flag("clang", "7", "14"), '-std=c++14')
-        self.assertEqual(cppstd_flag("clang", "7", "17"), '-std=c++17')
-        self.assertEqual(cppstd_flag("clang", "7", "20"), '-std=c++2a')
-
-        self.assertEqual(cppstd_flag("clang", "8", "11"), '-std=c++11')
-        self.assertEqual(cppstd_flag("clang", "8", "14"), '-std=c++14')
-        self.assertEqual(cppstd_flag("clang", "8", "17"), '-std=c++17')
-        self.assertEqual(cppstd_flag("clang", "8", "20"), '-std=c++2a')
+        with self.assertRaises(ConanException):
+            make_cppstd_default("gcc", "4.0")
+        self.assertEqual(make_cppstd_default("gcc", "5"), "gnu98")
+        self.assertEqual(make_cppstd_default("gcc", "6"), "gnu14")
+        self.assertEqual(make_cppstd_default("gcc", "6.1"), "gnu14")
+        self.assertEqual(make_cppstd_default("gcc", "7.3"), "gnu14")
+        self.assertEqual(make_cppstd_default("gcc", "8.1"), "gnu14")
+
+    def test_clang_make_cppstd_flags(self):
+        with self.assertRaises(ConanException):
+            make_cppstd_flag("clang", "2.0", "98")
+        with self.assertRaises(ConanException):
+            make_cppstd_flag("clang", "2.0", "gnu98")
+        with self.assertRaises(ConanException):
+            make_cppstd_flag("clang", "2.0", "11")
+        with self.assertRaises(ConanException):
+            make_cppstd_flag("clang", "2.0", "14")
+
+        with self.assertRaises(ConanException):
+            make_cppstd_flag("clang", "2.1", "98")
+        with self.assertRaises(ConanException):
+            make_cppstd_flag("clang", "2.1", "gnu98")
+        with self.assertRaises(ConanException):
+            make_cppstd_flag("clang", "2.1", "11")
+        with self.assertRaises(ConanException):
+            make_cppstd_flag("clang", "2.1", "14")
+
+        with self.assertRaises(ConanException):
+            make_cppstd_flag("clang", "3.0", "11")
+        with self.assertRaises(ConanException):
+            make_cppstd_flag("clang", "3.0", "14")
+
+        with self.assertRaises(ConanException):
+            make_cppstd_flag("clang", "3.1", "11")
+        with self.assertRaises(ConanException):
+            make_cppstd_flag("clang", "3.1", "14")
```

Comment:
Failing because of the helper function, not because of the underlying cppstd_flag being tested here

```suggestion
```

Reply:
removed